src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -257,9 +257,22 @@ async def Generate(
             )
 
         # TODO handle cancellation
-        result_generator: AsyncIterator[tuple[int, RequestOutput]] = (
-            merge_async_iterators(*generators)
-        )
+        result_generator: AsyncIterator[tuple[int, RequestOutput]]
+        try:
+            result_generator = merge_async_iterators(*generators)
+        except TypeError as exc:  # vllm > 0.5.4
```

Comment:
It's a matter of personal preference, but I think using `inspect` would make the code more concise.
```python
if "is_cancelled" in inspect.signature(merge_async_iterators).parameters:
```

Reply:
Agreed, even though this will be removed on the next release, that's cleaner. Done